pkgs/stdenv/generic/setup.sh

```diff
@@ -445,7 +445,7 @@ substituteAll() {
 
     # Select all environment variables that start with a lowercase character.
     # Will not work with nix attribute names (and thus env variables) containing '\n'.
```

Comment:
I was aware of that, see this comment. But in this case the `\n` was probably in the value, not the variable name. Wasn’t aware that set does something similar, though. It’s all a big mountain of :hankey:. @aszlig mentioned something about the regex being flaky as well.